src/Command/MakeUserCommand.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Command;
+
+use Doctrine\ORM\Mapping\Column;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Component\Console\Input\InputArgument;
+
+/**
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ * @author Ryan Weaver <weaverryan@gmail.com>
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+final class MakeUserCommand extends AbstractCommand
+{
+    protected static $defaultName = 'make:user';
+
+    public function configure()
+    {
+        $this
+            ->setDescription('Creates a the user doctrine class, the user provider and the user repository')
```

Comment:
Obviously, we want to make these commands as simple, and *not* interactive as possible (i.e. don't ask the user 10 questions). But, I think one question might make sense here:

> Do you need to store User details in the database?

This will tell us whether or not they need a `User` entity, or instead, a `User` model class. This would also change whether or not we generate a `UserProvider` (or just tell them to use the built-in `entity` provider). And finally, it will change what recommendations we give at the end: i.e. the exact changes they'll need to make to their `security.yml` file.

Reply:
We may also need to ask a second question... if the first question is "yes" to storing user information:

> Will you need to store an encoded password for each User in the database?

This would influence how we handle the `getPassword()` method and whether or not we have a `password` field. Sometimes, *even* when people have a local `User` entity, they validate their password in a different system (e.g. via a SSO API). This is very confusing to users, so it's a great opportunity to help generate the `User` class correctly based on this.